test/unit/lib/http_adapter/jquery_test.js

```diff
@@ -0,0 +1,209 @@
+        });
+
+        ['get', 'post', 'put', 'patch', 'delete'].forEach(function(method) {
+            describe('with http method: ' + method, function() {
+                it('makes ' + method + ' request using $.ajax', function() {
+                    return this.adapter.request(method, '/foo')
+                    .then(function(result) {
+                        this.$.ajax.calledOnce.should.be.true;
+                        this.$.ajax.lastCall.args[0].should.have.property('url');
+                        this.$.ajax.lastCall.args[0].url.should.contain('/foo');
+                        this.$.ajax.lastCall.args[0].should.have.property('method', method.toUpperCase());
+                    }.bind(this));
+                });
+
+                it('formats resolved ' + method + ' response', function() {
+                    this.$.ajax.returns(Promise.resolve({ body: 'some data' }));
+                    return this.adapter.request(method, '/foo')
+                    .then(function(value) {
+                        value.should.deep.equal({ body: 'some data'});
+                    });
+                });
+
+                it('sets content-type header', function() {
+                    return this.adapter.request(method, '/foo')
+                    .then(function(result) {
+                        var expectedContentType = 'application/json';
+                        this.$.ajax.lastCall.args[0].headers.should.have.property('Content-Type', expectedContentType);
+                    }.bind(this));
+                });
+
+                it('sets ajax header', function() {
+                    return this.adapter.request(method, '/foo')
+                    .then(function(result) {
+                        this.$.ajax.lastCall.args[0].headers.should.have.property('X-Requested-With', 'XMLHttpRequest');
+                    }.bind(this));
+                });
+
+                describe('timeout', function() {
+                    it('defaults to 30s', function() {
+                        return this.adapter.request(method, this.path)
+                        .then(function(result) {
+                            this.$.ajax.lastCall.args[0].should.have.property('timeout', 30000);
+                        }.bind(this));
+                    });
+
+                    it('can be overwritten', function() {
+                        this.adapter.setTimeout(5000);
+                        return this.adapter.request(method, this.path)
+                        .then(function(result) {
+                            this.$.ajax.lastCall.args[0].should.have.property('timeout', 5000);
+                        }.bind(this));
+                    });
+                });
+
+                describe('headers', function() {
+                    it('passes content-type header', function() {
+                        var expectedContentType = 'application/json';
+                        return this.adapter.request(method, this.path, this.options)
```

Comment:
Where is ```this.options``` set?  Maybe I'm just missing the reference somewhere.

Also it'd be nice to see that ```data``` attribute is set properly in the ajax call so we can see that options argument is tested properly in the call to ```this.$.ajax```.